pallets/gear/src/runtime_api.rs

```diff
@@ -211,8 +210,18 @@ where
             for note in journal {
                 core_processor::handle_journal(vec![note.clone()], &mut ext_manager);
 
-                if let Some(remaining_gas) = get_main_limit() {
-                    min_limit = min_limit.max(initial_gas.saturating_sub(remaining_gas));
+                match get_main_limit() {
+                    Some(remaining_gas) => {
+                        min_limit = min_limit.max(initial_gas.saturating_sub(remaining_gas))
+                    }
+                    None => match note {
+                        JournalNote::WaitDispatch { ref dispatch, .. }
```

Comment:
Add comment pls

Reply:
if from main chain, but finite wait (e.g. wait_for), it shouldn't result into initial gas

Reply:
wait calls with bounded duration would go to the first match arm. However if the calls have such big duration then it won't matter how the min_limit calculated: it will be the same

Reply:
> wait calls with bounded duration would go to the first match arm. However if the calls have such big duration then it won't matter how the min_limit calculated: it will be the same

oh, truth.
add comment anyway pls, it will be hard to understand wout context in future